---
title: "Using DESeq2 to perform differential expression analysis"
author: "J. Gibbons"
output: html_document
---

```{r load packages}
##Load packages. May need to install
library(DESeq2)
library(tidyverse)
```




```{r load data}
##Lets load the data
infile_feature_counts<-here::here("Rdata/vehicle_drug_feature_counts.txt")
df_counts<-read.delim(infile_feature_counts,comment.char = "#")
```


```{r}
##Lets remove the columns we don't need
df_counts<-df_counts %>% select(-Chr, -Start,-End,-Strand,-Length)

##To makes samples easier to read remove file extension from sample name
colnames(df_counts)<-gsub(pattern=".bam",replacement = "",
                                  x=colnames(df_counts),fixed=T)
```

```{r}
##Lets make a meta-data table (Required for DESeq2)
df_meta<-data.frame(Sample=colnames(df_counts)[2:5],
                    Info=colnames(df_counts)[2:5])

df_meta<-df_meta %>% separate(col=Info,into=c("Condition","Replicate"),sep="_")
```


```{r Creat_count_matrix}
##Put the count data into a matrix format so it can be used
##by DESeq2
m_counts<-as.matrix(df_counts[,2:ncol(df_counts)])
rownames(m_counts)<-df_counts$Geneid

##Make sure the colnames of the count data matches the sample
##names in the meta-data. Important because matching between
##count data and meta-data is position based
all(colnames(m_counts)==df_meta$Sample)
```


```{r Create_DESeq2_object}
##Create the DESeq2 data object
dds<-DESeqDataSetFromMatrix(countData = m_counts,
                            colData = df_meta,
                            design=~Condition)
```


```{r Sample_ordination}
##Perform sample ordination to assess sample quality
##PCA's generally don't work will with RNA-seq data
##The rlog function helps normalize the data 
##in a PCA friendly manner
rld<-rlog(dds)
DESeq2::plotPCA(rld,intgroup="Condition")
```


```{r Run_DESeq}
##Perform DESeq2 analysis to detect differences based on Condition
dds<-DESeq(dds)
```


```{r Extract_results}
##Extract the results
##Use contrast to specify the comparison to show and what the 
##reference is. drug is in the numerator and vehicle the
##denominator
##No filtering performed. Specify alpha for 
##optimizing independent filtering
res<-results(dds,contrast = c("Condition","drug","vehicle"),
             alpha=0.05,pAdjustMethod = "fdr")

df_res<-as.data.frame(res)
df_res$Transcript_ID<-rownames(df_res)
col_order<-c("Transcript_ID",colnames(df_res)[1:6])
df_res<-df_res[,col_order]
View(df_res)

##Split into Significantly up or down-regulated
df_res_sig_up<-subset(df_res,log2FoldChange>0 & padj<=0.1)
df_res_sig_down<-subset(df_res,log2FoldChange<0 & padj<=0.1)
```


```{r Get_normalized_data}
##Lets also get the normalized data for graphing
df_counts<-as.data.frame(counts(dds,normalized=T))
df_counts$Transcript_ID<-rownames(df_counts)
col_order<-c("Transcript_ID",colnames(df_counts[1:4]))
df_counts<-df_counts[,col_order]
```


```{r Write_out_results}
##Write out all of the results
write_tsv(df_res,here::here("Routput/all_deseq2_results.tsv"))
write_tsv(df_res_sig_down,here::here("Routput/sig_down_deseq2_results.tsv"))
write_tsv(df_res_sig_up,here::here("Routput/sig_up_deseq2_results.tsv"))
write_tsv(df_counts,here::here("Routput/deseq_normalized_counts.tsv"))
```
