```{r}
library(tidyverse)
```

```{r}
####Load the data####
infile<-here::here("Rdata/vehicle_drug_feature_counts.txt")

df_counts<-read.delim(infile,comment.char = "#")
```

```{r}
####Organize the data into a more convenient form####
##Get the transcript lengths
v_lengths<-df_counts$Length
names(v_lengths)<-df_counts$Geneid
##Convert the counts to a matrix
df_counts<-df_counts %>% select(-Chr, -Start,-End,-Strand,-Length)

##To makes samples easier to read remove file extension from sample name
colnames(df_counts)<-gsub(pattern=".bam",replacement = "",
                          x=colnames(df_counts),fixed=T)

m_counts<-as.matrix(df_counts[,2:ncol(df_counts)])
rownames(m_counts)<-df_counts$Geneid

##Make sure the data are aligned properly
all(names(v_lengths)==rownames(m_counts))
```

```{r}
####Calculate TPM####
v_length_per_kb<-v_lengths/1000
m_rpk<-m_counts/v_length_per_kb
v_pm<-colSums(m_rpk)/1000000

##This looks strange. We need to divide each sample by the 
##sequencing depth but R divides matrices by rows
##The easiest solution is to transpose the matrix
##so that the samples become the rows.
##We then transpose back to the usual form
m_tpm<-t(t(m_rpk)/v_pm) 
print(colSums(m_tpm)) #Columns should all sum to 1e+06
```

```{r}
#####Format data for write out####
df_tpm<-as.data.frame(m_tpm)
df_tpm$Gene_ID<-rownames(df_tpm)
df_tpm<-df_tpm %>% select(Gene_ID, everything())

write_tsv(df_tpm,here::here("Routput/tpm.tsv"))
```

```{r}
####Convert TPM matrix to longform
df_long<-pivot_longer(df_tpm,cols=2:ncol(df_tpm),
                      names_to="Sample",
                      values_to="TPM")

write_tsv(df_long,here::here("Routput/tpm_long_form.tsv"))
```